### PR TITLE
Revise page on security groups

### PR DESCRIPTION
**Pull request by geoffcline**
### Description

Revise the description of the security groups page. 

### Checklist
- n/a for docs updates

### CODE REVIEW

**Comment by geoffcline** [comment]
@oliviassss will you review this, please?

File: docs/deploy/security_groups.md [hunk 1]
```diff
@@ -1,32 +1,53 @@
...
-```
-name: k8s-traffic-<cluster_name>-<hash_of_cluster_name>
-tags: 
-    elbv2.k8s.aws/cluster: <cluster_name>
-    elbv2.k8s.aws/resource: backend-sg
-```
+- **Enable or Disable:** Use `--enable-backend-security-group` (default `true`) to toggle the shared backend security group.
 
-You can turn off the shared backend security group feature by setting `--enable-backend-security-group` to `false`. However, if you have a high number of Ingress resources with frontend security groups auto-generated by the controller, you might run into security group rule limits on the instance/ENI security groups.
+- **Specification:** If `--backend-security-group` (default empty) is left empty, an auto-generated security group with the following attributes will be created:
```
Comment (geoffcline) [inline docs/deploy/security_groups.md:31]: possibly change to "Default Behavior:" or add a heading like previous section.

File: docs/deploy/security_groups.md [hunk 2]
```diff
@@ -1,32 +1,53 @@
...
+
+### Default Behavior
+
+- If annotations are omitted, the Load Balancer Controller (LBC) initializes one security group for each load balancer.
+- The default setup permits traffic from `inbound-cidrs` to `listen-ports`.
 
 ## Backend Security Groups
 
-A single shared backend security group controls the traffic between load balancers and their target EC2 instances or ENIs. This security group is attached to the load balancers and is used as the traffic source in the ENI/Instance security group rules. The backend security group is shared between multiple load balancers.
+Backend Security Groups serve as the control mechanism for traffic between AWS Load Balancers and their target EC2 instances or ENIs. A single shared backend security group is utilized to manage the traffic.
```
Comment (geoffcline) [inline docs/deploy/security_groups.md:24]: bring back `This security group is attached to the load balancers and is used as the traffic source in the ENI/Instance security group rules.`

File: docs/deploy/security_groups.md [hunk 3]
```diff
@@ -1,32 +1,53 @@
...
+- **Auto-Creation:** When LBC auto-creates the frontend security group, it automatically adds rules allowing traffic from the load balancer to the backend instances/ENIs.
+- **Specification:** Use specific annotations on Ingress or Service resources for specifying frontend security groups, and manage the rules of instance/ENI security groups via additional annotations. 
 
-### Port Range Restrictions for Backend Security Group Rules
+### Port Range Restrictions
+From version v2.3.0 onwards, the controller applies restrictions to specific port ranges in the backend security group rules by default:
 
-As of version v2.3.0, the controller will by default restrict the backend security group rules to specific port ranges. You can set the controller flag `--disable-restricted-sg-rules` to `true` to get the backend security group rules to allow traffic to ALL ports.
```
Comment (geoffcline) [inline docs/deploy/security_groups.md:32]: link to how to set controller flags?

File: docs/deploy/security_groups.md [hunk 4]
```diff
@@ -1,32 +1,58 @@
...
+### Port Range Restrictions
 
-### Port Range Restrictions for Backend Security Group Rules
+From version v2.3.0 onwards, the controller restricts port ranges in the backend security group rules by default.
 
-As of version v2.3.0, the controller will by default restrict the backend security group rules to specific port ranges. You can set the controller flag `--disable-restricted-sg-rules` to `true` to get the backend security group rules to allow traffic to ALL ports.
+- Utilize the controller flag `--disable-restricted-sg-rules` set to `true` to permit traffic to all ports if needed.
```
Comment (johngmyers) [inline docs/deploy/security_groups.md:58]:
```suggestion
- If needed, set the controller flag `--disable-restricted-sg-rules` to `true` to permit traffic to all ports.
```

File: docs/deploy/security_groups.md [hunk 5]
```diff
@@ -1,32 +1,58 @@
...
+  - Be cautious when having numerous Ingress resources with auto-generated frontend security groups as it may lead to security group rule limits on the instance/ENI security groups.
+  - Ensure `--enable-backend-security-group` is set to true when specifying `alb.ingress.kubernetes.io/manage-backend-security-group-rules` or `service.beta.kubernetes.io/aws-load-balancer-manage-backend-security-group-rules`, to avoid errors.
 
 ### Management of Backend Security Group Rules
 
-When the LBC auto-creates the frontend security group for a load balancer, it automatically adds the security group rules to allow traffic from the load balancer to the backend instances/ENIs.
+Understand how to configure and manage security group rules effectively:
+
+- **Auto-Creation:** When LBC auto-creates the frontend security group, it automatically adds rules allowing traffic from the load balancer to the backend instances/ENIs.
+- **Specification:** Use specific annotations on Ingress or Service resources for specifying frontend security groups, and manage the rules of instance/ENI security groups via additional annotations.
```
Comment (johngmyers) [inline docs/deploy/security_groups.md:52]: This mangles the section to remove its entire purpose. The point is that specifying the frontend groups will disable LBC's behavior of adding security group rules to the security groups of targets. Furthermore, this behavior can be re-enabled with an annotation, which one depending on whether the resource is Ingress or Service.

**Review by johngmyers — CHANGES_REQUESTED** [review]

**Comment by geoffcline** [comment]
Thanks for the feedback @johngmyers ! I'm responding and revising now.

File: docs/deploy/security_groups.md [hunk 6]
```diff
@@ -1,32 +1,58 @@
-# Security groups
+# Security Groups for Load Balancers
 
-The AWS Load Balancer Controller classifies security groups into two categories: frontend and backend.
+Leverage security groups to limit client connections to your load balancers, and tightly define connections with nodes. The AWS Load Balancer Controller defines two classifications of security groups: **frontend** and **backend**.
+
+- **Frontend Security Groups:** Determine the clients that can access the load balancers.
```
[the remark this answers is not in the document]
Comment (geoffcline) [reply docs/deploy/security_groups.md:5]: I am trying a new approach for docs that includes more headings, bullets, use of bold, etc.

File: docs/deploy/security_groups.md [hunk 7]
```diff
@@ -1,32 +1,58 @@
...
+- **Backend Security Groups:** Regulate the traffic flowing between load balancers and their associated EC2 instances or ENIs.
```
[the remark this answers is not in the document]
Comment (geoffcline) [reply docs/deploy/security_groups.md:6]: I see your point here. I am intentionally including examples of targets. Users really struggle with understanding networking, and want to see network documentation improved. I am unsure if including more examples will help, but I see it as low risk.

File: docs/deploy/security_groups.md [hunk 8]
```diff
@@ -1,32 +1,58 @@
-# Security groups
+# Security Groups for Load Balancers
 
-The AWS Load Balancer Controller classifies security groups into two categories: frontend and backend.
+Leverage security groups to limit client connections to your load balancers, and tightly define connections with nodes. The AWS Load Balancer Controller defines two classifications of security groups: **frontend** and **backend**.
```
[the remark this answers is not in the document]
Comment (geoffcline) [reply docs/deploy/security_groups.md:3]: I changed to "restrict", and will elaborate on this under the backend heading.

**Pull request closed without merging**